python/pyarrow/_array.pyx

```diff
@@ -1053,18 +1053,21 @@ cdef maybe_coerce_datetime64(values, dtype, DataType type,
 
 
 
-def array(object sequence, DataType type=None, MemoryPool memory_pool=None):
+def array(object sequence, DataType type=None, MemoryPool memory_pool=None,
+          size=None):
     """
     Create pyarrow.Array instance from a Python sequence
 
     Parameters
     ----------
-    sequence : sequence-like object of Python objects
+    sequence : sequence-like or iterable object of Python objects.
+        If both type and size are specified may be a single use iterable.
```

Comment:
It might be nice to have a maxsize argument instead of an exact size with the interable (so we bail out in the event of infinitely long iterators)